### PR TITLE
Add QGIS 2.16.2

### PR DESCRIPTION
**Pull request by ceholden**
PR adds QGIS, a Qt based Geographic Information Systems (GIS) application

### CODE REVIEW

**Comment by ceholden** [comment]
Looks like Windows and Linux failed because of degraded Anaconda performance ([status](https://github.com/conda-forge/status/issues/16)) so I'll try again later. Not even building on Windows

Anyone have a clue as to the lint? Maybe my eyes are fuzzy but I only have one selector and looks correct:
``` yaml
  skip: true  # [not linux]
```
Maybe the comments in the requirements section?

**Comment by ceholden** [comment]
This line is being matched as a selector:

``` yaml
    - qgis --help || [[ "$?" == "2" ]]
```

Moved this test to `run_tests.sh` for now

File: recipes/qgis/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+requirements:
+  build:
+    - cmake
+    - flex
+    - bison
+    - m4
+    - python
+    - sip
+    - expat
+    - icu
```
Comment (ocefpaf) [inline recipes/qgis/meta.yaml]: `icu 56.*`

File: recipes/qgis/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    # Qt stack
+    - qt
+    - pyqt
+    - qscintilla2
+    - qca
+    - qjson
+    - qwt
+    - qwtpolar
+    # GDAL stack
+    - gdal >=2.1
```
Comment (ocefpaf) [inline recipes/qgis/meta.yaml]: `gdal 2.1.*`

File: recipes/qgis/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    - geos >=3.4
```
Comment (ocefpaf) [inline recipes/qgis/meta.yaml]: `geos 3.4.*`

**Comment by ceholden** [comment]
@ocefpaf Thanks for the review! I've updated the pinnings as you recommended -- I think you caught everything.

**Comment by ocefpaf** [comment]
Awesome recipe @ceholden! Thanks for this addition!!

**Pull request merged**